CHEWBBACA/utils/determine_paralogs.py

```diff
@@ -0,0 +1,549 @@
+
+    with open(output_file, write_mode) as out:
+        out.write(text+end_char)
+
+
+def join_list(lst, link):
+    """
+    """
+
+    return link.join(lst)
+
+
+def blast_inputs(clusters, output_directory):
+    """
+    """
+
+    ids_to_blast = []
+    for i in clusters:
+
+        cluster_file = os.path.join(output_directory,
+                                    '{0}_ids.txt'.format(i))
+        cluster_ids = [i] + [seq[0] for seq in clusters[i]]
+        cluster_lines = join_list(cluster_ids, '\n')
+        write_to_file(cluster_lines, cluster_file, 'w', '')
+        ids_to_blast.append(i)
+
+    return ids_to_blast
+
+
+def split_blast_inputs_by_core(blast_inputs, threads, blast_files_dir):
+    """
+    """
+
+    splitted_ids = [[] for cpu in range(threads)]
+    splitted_values = [[] for cpu in range(threads)]
+    cluster_sums = [0] * threads
+    i = 0
+    for cluster in blast_inputs:
+        cluster_file = os.path.join(blast_files_dir,
+                                    '{0}_ids.txt'.format(cluster))
+        with open(cluster_file, 'r') as infile:
+            cluster_seqs = [line.strip() for line in infile.readlines()]
+        splitted_values[i].append(len(cluster_seqs))
+        splitted_ids[i].append(cluster)
+        cluster_sums[i] += len(cluster_seqs)
+        i = cluster_sums.index(min(cluster_sums))
+
+    return [s for s in splitted_ids if len(s) > 0]
+
+
+def cluster_blaster(inputs):
+    """
+    """
+
+    blast_inputs = inputs[0:-4]
+    blastp_path = inputs[-4]
+    output_directory = inputs[-2]
+    proteins_file = inputs[-1]
+    blast_db = inputs[-3]
+
+    indexed_fasta = SeqIO.index(proteins_file, 'fasta')
+
+    for cluster in blast_inputs:
+
+        cluster_id = cluster
+        ids_file = os.path.join(output_directory,
+                                '{0}_ids.txt'.format(cluster_id))
+
+        with open(ids_file, 'r') as clstr:
+            cluster_ids = [l.strip() for l in clstr.readlines()]
+
+        fasta_file = os.path.join(output_directory,
+                                  '{0}_protein.fasta'.format(cluster_id))
+        # create file with protein sequences
+        get_sequences_by_id(indexed_fasta, cluster_ids, fasta_file)
+
+        blast_output = os.path.join(output_directory,
+                                    '{0}_blast_out.tsv'.format(cluster_id))
+        blast_command = ('{0} -db {1} -query {2} -out {3} '
+                         '-outfmt "6 qseqid sseqid score" '
+                         '-max_hsps 1 -num_threads {4} -evalue '
+                         '0.001 -seqidlist {5}'.format(blastp_path,
+                                                       blast_db,
+                                                       fasta_file,
+                                                       blast_output,
+                                                       1, ids_file))
+
+        os.system(blast_command)
+
+
+schema_short_dir = '/home/rfm/Desktop/ns_test/test_prep_schema2/short'
```

Comment:
This path needs to be changed or removed.

Reply:
@rfm-targa